conformance/utils/kubernetes/helpers.go

```diff
@@ -231,21 +231,88 @@ func parentsMatch(t *testing.T, expected, actual []v1alpha2.RouteParentStatus, n
 				return false
 			}
 		}
-		if len(aParent.Conditions) < len(eParent.Conditions) {
-			t.Logf("Expected more conditions to be present")
+		if !conditionsMatch(t, aParent.Conditions, eParent.Conditions) {
```

Comment:
expected should come before actual:
```suggestion
		if !conditionsMatch(t, eParent.Conditions, aParent.Conditions) {
```